OsmAnd/src/net/osmand/plus/routing/RouteCalculationResult.java

```diff
@@ -362,8 +362,40 @@ public double getRouteVisibleAngle() {
 					i--;
 				}
 			}
+
+			String streetName = null;
 			TurnType turn = s.getTurnType();
 
+			// Try to detect the case when a street changes is name to another (typically after road crossing)
+			// Then add RouteDirectionInfo of TurnType.straight()
+			if (turn == null && routeInd + 1 < list.size()) {
+				String curStreetName = s.getObject().getName(ctx.getSettings().MAP_PREFERRED_LOCALE.get(),
+						ctx.getSettings().MAP_TRANSLITERATE_NAMES.get());
+				String curIntRef = s.getObject().getValue("int_ref");
+				String curRoadRef2 = s.getObject().getValue("road_ref_2");
```

Comment:
road_ref_2, int_ref looks strange cause they are specific to some specific map area and it doesn't look as a general solution